pkg/devfile/generator/utils.go

```diff
@@ -157,12 +157,17 @@ func addSyncRootFolder(container *corev1.Container, sourceMapping string) string
 		{
 			Name:  EnvProjectsRoot,
 			Value: syncRootFolder,
-		}, {
-			Name:  EnvProjectsSrc,
-			Value: syncRootFolder,
 		},
 	}, container.Env...)
 
+	if sourceMapping != "" {
+		container.Env = append(container.Env,
+			corev1.EnvVar{
+				Name:  EnvProjectsSrc,
+				Value: sourceMapping,
```

Comment:
Out of curiosity, why do we only want to set `EnvProjectsSrc` if `sourceMapping` is set? Since if `sourceMapping` is not set then we assign `syncRootFolder = DevfileSourceVolumeMount` is that not the source we would need? Not saying it is wrong I just want to understand how it is working because if I am not mistaken we aren't setting `EnvProjectsSrc` if `sourceMapping` is unset

Reply:
`I think here it should take the source value not the root.`
Ah, I implemented it as described, but I think I misunderstood something.

Reply:
@thepetk do you have context related to your comment here? https://github.com/devfile/library/pull/220#discussion_r1706961576

Reply:
the main problem is the we get the same identical values ? which should not be there right ? we doesn't append if the values are same am i correct ?

Reply:
> @thepetk do you have context related to your comment here? https://github.com/devfile/library/pull/220#discussion_r1706961576

From what I see, we're assigning everytime the root dir as project source. However I'd expect source to be `PROJECT_ROOT` + `RELATIVE_SOURCE_PATH`.

Now, following the flow of the library both source & root are set [here](https://github.com/devfile/library/blob/main/pkg/devfile/generator/utils.go#L719). So, in case no project has been found there's no need for `PROJECT_SOURCE` path. In case we'd like to set `PROJECT_SOURCE` no matter what I'd expect updating (suggestion):

1. The [addSyncFolder](https://github.com/devfile/library/blob/main/pkg/devfile/generator/utils.go#L168-L199) function to include this case (inc test cases updates).
2. The [getAllContainers](https://github.com/devfile/library/blob/main/pkg/devfile/generator/utils.go#L689-L734) function by removing the early return for no-projects-found cases and this way we set `PROJECT_SOURCE` everytime.

My main question here is if it's ok to add a value for `PROJECT_SOURCE` even if we haven't found anything. That said I'd expect a null value for this env var for such cases.

Reply:
Better to resolve in favor of https://github.com/devfile/library/pull/220/files#r1717510608 and we can continue the discussion there :)